### PR TITLE
refactor: unify return types of clients

### PR DESCRIPTION
**Pull request by warber**
#### What this PR does / Why we need it:
Although achieving uniformity across all clients' interfaces may not be practical, standardizing the return signature can be easily accomplished. This simplifies the usage of the client(s) and enhances convenience.
Moreover, these changes support the flexibility to not break any code once an API suddenly introduces pagination.

#### Does this PR introduce a user-facing change?
:exclamation: Return types are different, thus clients need to adopt the usage.

### CODE REVIEW

File: api/clients/automation/automation.go [hunk 1]
```diff
@@ -28,9 +28,7 @@ import (
 	"strconv"
 )
 
-type Response struct {
-	api.Response
-}
+type Response = api.Response
```
Comment (jskelin) [inline api/clients/automation/automation.go:31]: Is it possible to remove the alias also?
I don't see added value

File: api/clients/buckets/bucket.go [hunk 2]
```diff
@@ -32,22 +32,18 @@ import (
...
-}
-
 type response struct {
 	api.Response
 	BucketName string `json:"bucketName"`
 	Status     string `json:"status"`
 	Version    int    `json:"version"`
 }
 
+type Response = api.Response
```
Comment (jskelin) [inline api/clients/buckets/bucket.go:42]: Is it possible to remove the alias also?
I don't see added value

File: api/clients/buckets/bucket.go [hunk 3]
```diff
@@ -32,22 +32,18 @@ import (
...
+
 // ListResponse is a Bucket API response containing multiple bucket objects.
 // For convenience, it contains a slice of Buckets in addition to the base api.Response data.
-type ListResponse struct {
-	api.ListResponse
-}
+type ListResponse = api.PagedListResponse
```
Comment (jskelin) [inline api/clients/buckets/bucket.go:46]: Is it possible to remove the alias also?
I don't see added value

**Review by jskelin — APPROVED** [review]

File: api/clients/buckets/bucket.go (hunk 2 above)
Comment (warber) [reply api/clients/buckets/bucket.go:42]: The advantage is, that you don't need to import two packages but just e.g. "github.com/dynatrace/dynatrace-configuration-as-code-core/api/clients/buckets" when using that client.

**Review by Laubi — APPROVED** [review]

**Pull request merged**